gateway/ln-gateway/src/lib.rs

```diff
@@ -256,6 +257,21 @@ pub struct Gateway {
     channel_id_generator: Arc<Mutex<AtomicU64>>,
```

Comment:
A `Mutex` around an atomic type seems quite useless.

Reply:
Removed `AtomicU64` and replaced with `u64`. Also added overflow logic when registering a federation and removed the long standing/annoying TODO.